lib/ipam/ipam.go

```diff
@@ -1383,6 +1344,62 @@ func (c ipamClient) convertBackendToIPAMConfig(cfg *model.IPAMConfig) *IPAMConfi
 	}
 }
 
+// maybeReleaseAffinity retrieves the pool and node for the given block and determines
+// if the pool still selects node. If it no longer matches, it will release the block
+// affinity for that node.
+// Returns a bool indicating if the block was released.
+func (c ipamClient) maybeReleaseAffinity(ctx context.Context, b *model.AllocationBlock) (bool, error) {
+	logCtx := log.WithField("cidr", b.CIDR)
+	// Retrieve node for this allocation. We do this so we can clean up affinity for blocks
+	// which should no longer be affine to this host.
+	host := getHostAffinity(b)
+	logCtx.Debugf("block affinity: %s", host)
```

Comment:
Instead of logging this out, let's add it to the context logger.

It would be nice to have as context for other logs we emit later on.